src/Form/Form.php

```diff
@@ -595,6 +613,120 @@ protected function deleteMissingQuestions(): void
         unset($this->input['_found_questions']);
     }
 
+    /**
+     * Update form's comments using the special data found in
+     * $this->input['_comments']
+     *
+     * @return void
+     */
+    protected function updateComments(): void
+    {
+        $comments = $this->input['_comments'] ?? [];
+
+        // Keep track of comments found
+        $found_comments = [];
+
+        // Parse each submitted comment
+        foreach ($comments as $comment_data) {
+            $comment = new Comment();
+
+            if ($comment_data["_use_uuid_for_sections_id"]) {
+                // This question was added to a newly created section
+                // We need to find the correct section id using the temporary UUID
+                $uuid = $comment_data['forms_sections_id'];
+                $comment_data['forms_sections_id'] = $_SESSION['form_editor_sections_uuid'][$uuid] ?? 0;
+            }
```

Comment:
This is the kind of logic that will be hard to maintain. It requires additionnal inputs on the form side and a temporary storage in session data.

Maybe it could be handled in a simplier way by nesting data in the POST payload, for instance
```
_sections[0]=[...]
_comments[0]=[...]
```
may be changed to 
```
_sections[0]=[
    ...
    _comments[0]=[...]
]
```

The comment would be part of the section payload and it would be therefore easier to attach it to the section. Also, the payload would probably be easier to read.

As this logic is already used by questions, it is not a blocker for the current PR, but I think it could be interesting to simplify this.

Reply:
This was the original behavior in the first forms iteration (nested arrays).
There was some issues with it but I don't remember exactly what.

I agree the uuid stuff is not ideal too, a better solution would be welcome.